### PR TITLE
Adds more verbose help function to service run

### PR DESCRIPTION
**Pull request by ed-**
"serviced service run <svc> help" displays commands available to that service.

Additionally, if a Command has a Description defined, the help message will
display that in the "Available commands" listing.

Closes CC-1102

### CODE REVIEW

File: cli/cmd/service.go [hunk 1]
```diff
@@ -286,6 +286,43 @@ func (c *ServicedCli) printServicesAll(ctx *cli.Context) {
 	}
 }
 
+func (c *ServicedCli) printHelpForRun(svc *service.Service, command string) (returncode int) {
+	found := false
```
Comment (iancmcc) [inline cli/cmd/service.go]: var found bool

File: cli/cmd/service.go [hunk 2]
```diff
@@ -286,6 +286,43 @@ func (c *ServicedCli) printServicesAll(ctx *cli.Context) {
...
+	availablecommands := make([]string, 0, len(svc.Commands))
+	sort.Strings(availablecommands)
```
Comment (iancmcc) [inline cli/cmd/service.go]: This line, while not expensive, is also not particularly useful.

File: cli/cmd/service.go [hunk 3]
```diff
@@ -286,6 +286,43 @@ func (c *ServicedCli) printServicesAll(ctx *cli.Context) {
...
+		fmt.Printf("Available commands for %v:\n", svc.Name)
+		for _, commandname := range availablecommands {
+			fmt.Printf("    %-20v  %v\n", commandname, svc.Commands[commandname].Description)
+		}
+		if len(availablecommands) == 0 {
+			fmt.Println("    No commands available.")
+		}
+		return 1
+
+	} else {
```
Comment (iancmcc) [inline cli/cmd/service.go]: Go style prefers that you simply end the if block here and return -1 in the function scope.

File: cli/cmd/service.go [hunk 4]
```diff
@@ -977,6 +1014,11 @@ func (c *ServicedCli) cmdServiceRun(ctx *cli.Context) error {
 		return c.exit(1)
 	}
 
+	returncode := c.printHelpForRun(svc, args[1])
```
Comment (iancmcc) [inline cli/cmd/service.go]: if returncode := c.printHelpForRun(svc, args[1]); returncode >= 0 {
    return c.exit(returncode)
}

File: domain/service/json.go [hunk 5]
```diff
@@ -40,6 +40,7 @@ func (s *Service) UnmarshalJSON(b []byte) error {
 			s.Commands[k] = domain.Command{
 				Command:         v,
 				CommitOnSuccess: true,
+				Description:     "",
```
Comment (iancmcc) [inline domain/service/json.go]: This is unnecessary.

File: domain/servicedefinition/servicedefinition.go [hunk 6]
```diff
@@ -185,6 +185,7 @@ func (s *ServiceDefinition) UnmarshalJSON(b []byte) error {
 			s.Commands[k] = domain.Command{
 				Command:         v,
 				CommitOnSuccess: true,
+				Description:     "",
```
Comment (iancmcc) [inline domain/servicedefinition/servicedefinition.go]: This is similarly unnecessary.

File: cli/cmd/service.go [hunk 7]
```diff
@@ -286,6 +286,46 @@ func (c *ServicedCli) printServicesAll(ctx *cli.Context) {
 	}
 }
 
+func (c *ServicedCli) printHelpForRun(svc *service.Service, command string) (returncode int) {
+	var (
+		found             bool
+		availablecommands []string
+	)
+	found = false
+	availablecommands = make([]string, 0, len(svc.Commands))
```
Comment (iancmcc) [inline cli/cmd/service.go]: You don't need these two lines anymore.

**Comment by tgjonestx** [comment]
ok to test

**Comment by tgjonestx** [comment]
test this please

**Pull request closed without merging**